src/shared/hooks/useDisablePictureInPicture.tsx

```diff
@@ -0,0 +1,20 @@
+import { type RefObject, useEffect } from 'react';
+
+export const useDisablePictureInPicture = (ref: RefObject<HTMLElement>) => {
+	useEffect(() => {
+		if (ref.current) {
+			observer.observe(ref.current, {
+				childList: true,
+				subtree: true,
+			});
+		}
+
+		return () => observer.disconnect();
```

Comment:
Does this also work?
```suggestion
		return observer.disconnect;
```